code/model/Translatable.php

```diff
@@ -612,13 +612,19 @@ function augmentSQL(SQLQuery &$query, DataQuery &$dataQuery = null) {
 			// the query contains this table
 			// @todo Isn't this always the case?!
 			&& array_search($baseTable, array_keys($query->getFrom())) !== false 
-			// or we're already filtering by Lang (either from an earlier augmentSQL() 
-			// call or through custom SQL filters)
-			&& !preg_match('/("|\'|`)Locale("|\'|`)/', implode(' ', $query->getWhere()))
 			//&& !$query->filtersOnFK()
 		)  {
-			$qry = sprintf('"%s"."Locale" = \'%s\'', $baseTable, Convert::raw2sql($locale));
-			$query->addWhere($qry); 
+			// Or we're already filtering by Lang (either from an earlier augmentSQL() 
+			// call or through custom SQL filters)
+			$filtersOnLocale = array_filter($query->getWhere(), function($predicates) {
+				foreach($predicates as $predicate => $params) {
```

Comment:
Ok, to answer the question and make a comment; the $predicates in this case would be one of;
- a string (3.1) containing the whole condition
- an array with a single item in `array('condition' => array('params'))` format.

Simple way to extract the string condition is.

```php
protected function filtersOnLocale($query) {
    foreach($query->getWhere() as $condition) {
        // Compat for 3.1/3.2 where syntax
        if(is_array($condition)) {
            reset($condition);
            $condition = key($condition);
        }
        if(preg_match('/("|\'|`)Locale("|\'|`)/', $condition)) return true;
    }
}
````